### PR TITLE
Fix Flyout Footer measuring and Content Margin

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change ###

- Open up a few renderer APIs to make it easier to customize parts of the Shell Flyout
- Fix measuring on the footer to pass in double.Infinity instead of -1. 
- Set the bottom margin on the content inside the flyout so it doesn't flow behind the footer view

### API Changes ###

Added:
 - protected virtual ShellFlyoutTemplatedContentRenderer.ShellContext
 - protected virtual ShellFlyoutTemplatedContentRenderer.FooterView
 - protected virtual ShellFlyoutTemplatedContentRenderer.View
 - protected virtual ShellFlyoutTemplatedContentRenderer.UpdateFlyoutHeader()
 - protected virtual ShellFlyoutTemplatedContentRenderer.UpdateFlyoutFooter()
 - protected virtual ShellItemRenderer.UpdateTabBarVisibility()

### Platforms Affected ### 
- Android

### Testing Procedure ###
- UI Test included

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs [hunk 1]
```diff
@@ -36,37 +36,49 @@ public View View
...
-				_renderer.View.Measure(empty, empty);
+				Renderer.View.Measure(empty, empty);
 				return;
 			}
 
 			var request = View.Measure(width, height, MeasureFlags.None);
 
 			var layoutParams = NativeView.LayoutParameters;
-			if (height == -1)
+			if (height == -1 || double.IsInfinity(height))
```
Comment (pictos) [inline Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs:70]: We're setting the value to `PositiveInfinite`, do we need the check for `-1`?

File: Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs [hunk 2]
```diff
@@ -36,37 +36,49 @@ public View View
...
 				height = request.Request.Height;
 
-			if (width == -1)
+			if (width == -1 || double.IsInfinity(width))
```
Comment (pictos) [inline Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs:73]: Same here

File: Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs (hunk 1 above)
Comment (PureWeen) [reply Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs]: 👍

File: Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs (hunk 2 above)
Comment (PureWeen) [reply Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs]: 👍

File: Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs [hunk 3]
```diff
@@ -36,37 +36,49 @@ public View View
...
-				_renderer.View.Measure(empty, empty);
+				Renderer.View.Measure(empty, empty);
 				return;
 			}
 
 			var request = View.Measure(width, height, MeasureFlags.None);
 
 			var layoutParams = NativeView.LayoutParameters;
-			if (height == -1)
+			if (double.IsInfinity(height))
```
Comment (jsuarezruiz) [inline Xamarin.Forms.Platform.Android/Renderers/ShellViewRenderer.cs:70]: :heart:

**Review by jsuarezruiz — APPROVED** [review]

**Review by rmarinho — APPROVED** [review]

**Pull request merged**